src/science/roughness/cable_roughness.F90

```diff
@@ -35,19 +34,100 @@ MODULE cable_roughness_module
 USE cable_phys_constants_mod, ONLY : CCSW    => CSW
 USE cable_phys_constants_mod, ONLY : CGRAV   => GRAV
 USE cable_other_constants_mod, ONLY : CLAI_THRESH => LAI_THRESH 
-   
+
+!*# Overview
+!
+! The procedures contained in this module calculate the roughness parameters
+! and the aerodynamic contribution to the resistances controlling the fluxes 
+! of momentum, heat and water vapour between the land and atmosphere for each 
+! land point.
+!
+! The formulations take into account vegetation and snow cover. The dependence ! on atmospheric conditions (surface heat fluxes) is incorporated later within ! the [[cable_canopy_module]].
+! 
+!## Scientific description
+!
+! The scientific basis for the formulae is a combination of Localised Near 
+! Field theory for aerodynamic transfer within canopies (for the heat and 
+! water vapour fluxes) and bulk formulae for the roughness length and 
+! displacement height accounting for roughness sublayer effects (for the 
+! momentum flux).
+!
+! The calculations take into account whether:
+! 
+! 1. the soil model used is SLI or the default (soilsnow)
+! 2. the land point is vegetated (LAI > LAI_THRESH) or not
+!
+! The primary references for these formulations are
+!
+! * [Raupach MR (1989)](https://doi.org/10.1002/qj.49711548710)
+! * [Raupach MR (1992)](https://doi.org/10.1007/BF00155203)
+! * [Raupach MR (1994)](https://doi.org/10.1007/BF00709229)
+! * [Kowalczyk et al. (2006)](http://www.cmar.csiro.au/e-print/open/kowalczykea_2006a.pdf)
+!
+! <br></br>
+! The principal outputs from the MODULE are
```

Comment:
I would argue the module does not output anything, that's the subroutine and I would move all this to the subroutine level.

Reply:
Happy to go with the structure/language of the broader documentation.